src/FixedSizeArrays.jl

```diff
@@ -173,4 +225,126 @@ function Base.reshape(a::FixedSizeArray{T}, size::NTuple{N,Int}) where {T,N}
     FixedSizeArray{T,N}(Internal(), a.mem, size)
 end
 
+# `collect_as`
+
+function collect_as_fsa0(iterator, ::Val{nothing})
+    x = only(iterator)
+    ret = FixedSizeArray{typeof(x),0}(undef)
+    ret[] = x
+    ret
+end
+
+function collect_as_fsa0(iterator, ::Val{E}) where {E}
+    E::Type
+    x = only(iterator)
+    ret = FixedSizeArray{E,0}(undef)
+    ret[] = x
+    ret
+end
+
+function fill_fsa_from_iterator!(a, iterator)
+    actual_count = 0
+    for e ∈ iterator
+        actual_count += 1
+        a[actual_count] = e
+    end
+    if actual_count != length(a)
+        throw(ArgumentError("`size`-`length` inconsistency"))
```

Comment:
This line isn't covered yet, maybe I get to that later, not sure if that's important enough to block merging.

Reply:
Perhaps the check could even be removed.

Reply:
I'd prefer to cover error paths as well, if possible :slightly_smiling_face: